docs/deep-dives/github.md

```diff
@@ -43,7 +43,37 @@ steps:
 ```
 
 With Docker Compose,
-declare the environment variables and volume mounts in your [Compose file]:
+declare volume mounts in your [Compose file].
+
+You won't need to declare the environment variables in the compose file since `propagate-environment: true` will ensure the variables defined are propagated to the docker environment.
+
+For example, with the [Docker Compose Buildkite plugin], we will need to add the `propagate-environment: true`:
+
+```yaml
+steps:
+  - commands:
+      - pnpm run lint
+      - pnpm run test
+    env:
+      # At SEEK, this instructs the build agent to populate the GITHUB_API_TOKEN environment variable for this step.
+      GET_GITHUB_TOKEN: 'please'
+    plugins:
+      - *aws-sm
+      - *private-npm
+      - *docker-ecr-cache
+      - docker#v5.0.0:
+          # Enable GitHub integrations.
+          environment:
+            - GITHUB_API_TOKEN
+          propagate-environment: true
+          volumes:
+            # Mount cached dependencies.
+            - /workdir/node_modules
+```
+
+And then your [Compose file] will not need to define the `environment` like you had to before to ensure the environment variables are available in Docker
+
+Docker compose file before:
```

Comment:
Hmmm. I don't think we have to document the before and after within the deep-dive itself. We can probably get away with just showing what the optimal configuration is.

We can probably move some of the `before` and `after` scenarios to the changeset